src/rest-server/src/middlewares/v2/protocol.js

```diff
@@ -107,6 +128,18 @@ const protocolValidate = (protocolYAML) => {
 };
 
 const protocolRender = (protocolObj) => {
+  // render auth for Docker image
+  for (let name of Object.keys(protocolObj.prerequisites.dockerimage)) {
+    if ('auth' in protocolObj.prerequisites.dockerimage[name]) {
+      for (let prop of Object.keys(protocolObj.prerequisites.dockerimage[name].auth)) {
+        protocolObj.prerequisites.dockerimage[name].auth[prop] = render(
+          protocolObj.prerequisites.dockerimage[name].auth[prop],
+          {'$secrets': protocolObj.secrets},
```

Comment:
you assume the auth field must use secrets?

Reply:
no, the value can be null or undefined